### PR TITLE
Pass the width constraint to the text layout engine if ellipsizing is enabled

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@Hixie

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 1]
```diff
@@ -148,7 +148,10 @@ class RenderParagraph extends RenderBox {
   }
 
   void _layoutText({ double minWidth: 0.0, double maxWidth: double.INFINITY }) {
-    _textPainter.layout(minWidth: minWidth, maxWidth: _softWrap ? maxWidth : double.INFINITY);
+    _textPainter.layout(
+        minWidth: minWidth,
+        maxWidth: _softWrap || _overflow == TextOverflow.ellipsis ? maxWidth : double.INFINITY
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/paragraph.dart]: fwiw, i'd probably use the getters rather than the fields, mostly for readability. your call.

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
It's hard to tell but this may have caused microbenchmarks stock_layout_iteration to regress a little.

**Pull request merged**